File: src/main/java/com/focus/focus/service/StatsService.java
package com.focus.focus.service;

import com.focus.focus.dto.StatsDataPoint;
import com.focus.focus.dto.TopTagDto;
import com.focus.focus.dto.TopTaskDto;
import com.focus.focus.model.entity.Usuario;

import java.util.List;

public interface StatsService {

    List<StatsDataPoint> getWeeklyStats(Usuario usuario);

    List<StatsDataPoint> getMonthlyStats(Usuario usuario);

    List<TopTaskDto> getTopTasks(Usuario usuario, int limit);

    List<TopTagDto> getTopTags(Usuario usuario, int limit);
}

File: frontend/src/services/statsService.ts
import { apiRequest } from "./api";

export interface StatsDataPoint {
  label: string;
  value: number;
}

export interface TopTaskDto {
  tareaId: number;
  titulo: string;
  pomodorosCompletados: number;
}

export interface TopTagDto {
  etiqueta: string;
  totalPomodoros: number;
}

export const statsService = {
  async getWeeklyStats(): Promise<StatsDataPoint[]> {
    return apiRequest<StatsDataPoint[]>("/api/stats/weekly");
  },

  async getMonthlyStats(): Promise<StatsDataPoint[]> {
    return apiRequest<StatsDataPoint[]>("/api/stats/monthly");
  },

  async getTopTasks(limit = 5): Promise<TopTaskDto[]> {
    return apiRequest<TopTaskDto[]>(`/api/stats/top-tasks?limit=${limit}`);
  },

  async getTopTags(limit = 5): Promise<TopTagDto[]> {
    return apiRequest<TopTagDto[]>(`/api/stats/top-tags?limit=${limit}`);
  },
};
